### PR TITLE
Cleanup assemblyinfo

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Remove old code from AssemblyInfo.cs

### Additional considerations

We could use `IgnoreAccessChecksToAttribute` instead of `InternalsVisibleToAttribute`.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by iSazonov** [comment]
Seems Appveyor CI temporary failed.

**Comment by anmenaga** [comment]
Restarted AppVeyor build.

File: src/Microsoft.Management.Infrastructure.CimCmdlets/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,11 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-using System.Diagnostics;
-using System.Reflection;
-using System.Runtime.CompilerServices;
-
-[assembly:InternalsVisibleTo("Microsoft.Windows.DSC.CoreConfProviders,PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```
Comment (anmenaga) [inline src/Microsoft.Management.Infrastructure.CimCmdlets/AssemblyInfo.cs]: how do you know that this is not used?

Comment (iSazonov) [reply src/Microsoft.Management.Infrastructure.CimCmdlets/AssemblyInfo.cs]: I don't know. It is for MSFT review.
I believe that internal and private visibility is permissible only in the repo. All other should use only public APIs or `IgnoreAccessChecksToAttribute` if needed.

Comment (TravisEz13) [reply src/Microsoft.Management.Infrastructure.CimCmdlets/AssemblyInfo.cs]: I'll email DSC and ask.

Comment (TravisEz13) [reply src/Microsoft.Management.Infrastructure.CimCmdlets/AssemblyInfo.cs]: I verified the DSC team still needs this.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
Add DSC intervals visible to back

**Comment by iSazonov** [comment]
@TravisEz13 Will do.
Could you please share why they need the visibility?
If they use it only for tests with precompiled dlls from the repo and use Roslyn they could migrate to using `IgnoreAccessChecksToAttribute`. Of cause if they still use old test system we have to keep the visibility for them.

**Comment by adityapatwardhan** [comment]
@iSazonov  Please push a commit with `[Feature]` tag to execute all tests.

**Comment by iSazonov** [comment]
@TravisEz13 Please update your review.

**Comment by TravisEz13** [comment]
They do use the visibility for testing, but some of it is legacy.

File: src/System.Management.Automation/AssemblyInfo.cs [hunk 2]
```diff
@@ -12,33 +12,19 @@
...
-[assembly: InternalsVisibleTo(@"Microsoft.PowerShell.ISECommon" + @",PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
-[assembly: InternalsVisibleTo(@"Microsoft.PowerShell.Editor" + @",PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
-[assembly: InternalsVisibleTo(@"powershell_ise" + @",PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```
Comment (TravisEz13) [inline src/System.Management.Automation/AssemblyInfo.cs]: Do we know why this was here?

Comment (iSazonov) [reply src/System.Management.Automation/AssemblyInfo.cs]: They all relate to ISE. I think we can safely remove this.

**Review by adityapatwardhan — APPROVED** [review]

File: src/System.Management.Automation/AssemblyInfo.cs (hunk 2 above)
Comment (adityapatwardhan) [reply src/System.Management.Automation/AssemblyInfo.cs]: They were added when we had a full CLR build as well.

**Comment by iSazonov** [comment]
@TravisEz13 Please update your review.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**